### Задача 1
*Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.*


Дано:
* $\sigma=16$
* $\alpha=1-0.95=0.05$
* $\Mu = 80$
* $n=256$

Применим формулу доверительного интервала при известном СКО генеральной совокупности:
$$\bar{X} \pm Z_{\frac{\alpha}{2}}*\frac{\sigma}{\sqrt{n}}$$
$\bar{X} = \Mu = 80$  
$Z_{\frac{\alpha}{2}}$ по таблице равен 1.96
$$80 \pm 1.96*\frac{16}{\sqrt{256}}=80\pm 1.96*1$$

Ответ: интервал с надежностью в 95% будет  **[78,04 ; 81,96]**

И то же самое в Питоне:

In [9]:
import scipy.stats as stats
import numpy as np
def Trusted_intZ(n,sigma,mid,alpha):
    Z_a=stats.norm.ppf(1-alpha/2)
    dev=np.abs(Z_a*sigma/np.sqrt(n))
    return f'[{mid-dev};{mid+dev}]'

Trusted_intZ(256,16,80,0.05)

'[78.04003601545995;81.95996398454005]'

### Задача 2

*В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:*

    6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1  
    
*Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.*

Здесь среднее квадратичное отклонение ГС мы не знаем, поэтому будем считать его по выборке и использовать криттерий Стьюдента.
$$ \bar{X} \pm t_{\frac{\alpha}{2}}*\frac{\sigma_x}{\sqrt{n}} $$

Посчитаем среднее арифметическое и стандартное отклонение через питон:

In [18]:
x_list = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
np.std(x_list)

0.42766809560686203

In [15]:
np.mean(x_list)

6.590000000000001

По таблице $t_{\frac{\alpha}{2}}$ со степенями свободы 9 и вероятностью 0.975 будет равно 2.262  
Подставим все это в формулу:
$$6.59 \pm 2.262*\frac{0.43}{\sqrt{10}}=6.59 \pm 2.262*0.136=6.59 \pm 0.308$$
Ответ: доверительный 95-% интервал будет **[6.282 ; 6.898]**

А здесь проверка через функцию с нашей формулой в Питоне:

In [19]:

def Trusted_intT(l:list,alpha:float)->str:
    n=len(l)
    Z_t=stats.t.ppf(1-alpha/2, n-1)
    sigma= np.std(l)
    mid = np.mean(l)
    dev=np.abs(Z_t*sigma/np.sqrt(n))
    return f'[{mid-dev};{mid+dev}]'


Trusted_intT(x_list,0.05)

'[6.284064674352047;6.8959353256479545]'

Совпало, с некоторой погрешностью в тысячных

### Задача 3
*На препарате А положительный результат лечения наблюдается у 17 из 32 пациентов, а на препарате В у 9 из 22. Построить 95% доверительный интервал для разности долей.
Обнаружены ли статистически значимые различия?*

Формула для сравнения долей:

$$z = \frac{\bar{p_1}-\bar{p_2}}{\sqrt{p(1-p)(\frac{1}{n_1}+\frac{1}{n_2})}} \text{ ,где  }p = \frac{m_1+m_2}{n_1+n_2}$$
$\bar{p_1}=17/32=0.53125$  
$\bar{p_2}=9/22=0.409$  
Подставим значения из условия, где в числителе $p$ положительные результаты лечения в выборках, а в знаменателе число пациентов каждой выборки.  
$$p = \frac{17+9}{22+32}=0.481$$
Так как доверительный интервал мы строим по непрерывным величинам, мы вводим в числитель $z$ еще поправку на непрерывность:
$$z = \frac{|\bar{p_1}-\bar{p_2}|-\frac{1}{2}(\frac{1}{n_1}+\frac{1}{n_2})}{\sqrt{p(1-p)(\frac{1}{n_1}+\frac{1}{n_2})}}$$
$$z = \frac{|0.53125-0.409|-\frac{1}{2}(0.03125+0.045)}{\sqrt{0.481*(1-0.481)(0.03125+0.045)}}$$
$$z = \frac{0.12225-0.038125}{0.138}=\frac{0.084125}{0.138}=0.609$$


Это меньше оценки статистической   значимости при $\alpha=5\%$, и $Z_{\alpha/2}=1.96$, значит статистически значимых различий между двумя препаратами нет. 

Теперь построим доверительный 95%-й интервал для разности долей:  
$$\Delta \pm z_{\frac{\alpha}{2}}*S_{\Delta} \text{ ,где  } S_{\Delta}=\sqrt{p(1-p)((\frac{1}{n_1}+\frac{1}{n_2}))}\text{,  то есть знаменатель, найденный нами ранее в формуле z-криттерия}$$
$$\Delta = \bar{p_1}-\bar{p_2} = 0.12225 $$
 $ 0.12225 \pm 1.96*0.138 =  0.12225 \pm 0.27048$  
 **Доверительный интервал: [-0,148; 0,392]**